Libplanet/Net/Swarm.cs

```diff
@@ -2003,21 +2033,43 @@ private void TransferRecentStates(GetRecentStates getRecentStates)
                 {
                     Guid chainId = BlockChain.Id;
 
+                    _logger.Debug(
+                        "Getting state references from {Offset}",
+                        getRecentStates.Offset);
+
+                    long lowestIndex =
+                        (@base is HashDigest<SHA256> bbh &&
+                         _store.GetBlockIndex(bbh) is long bbIdx)
+                            ? bbIdx + getRecentStates.Offset
+                            : 0 + getRecentStates.Offset;
+                    long targetIndex =
+                        (target is HashDigest<SHA256> tgt &&
+                         _store.GetBlockIndex(tgt) is long tgtIdx)
+                            ? tgtIdx
+                            : long.MaxValue;
+                    long highestIndex = lowestIndex + FindNextStatesChunkSize - 1 > targetIndex
+                        ? targetIndex
+                        : lowestIndex + FindNextStatesChunkSize - 1;
+
+                    nextOffset = highestIndex == targetIndex
+                        ? -1
+                        : getRecentStates.Offset + FindNextStatesChunkSize;
+
                     stateRefs = _store.ListAllStateReferences(
                         chainId,
-                        onlyAfter: @base,
-                        ignoreAfter: target
+                        lowestIndex: lowestIndex,
+                        highestIndex: highestIndex
```

Comment:
How about leaving only one API for `ListAllStateReferences`?
If we use the previous one this code can be
```
                    var lowestBlockHash = lowestIndex == 0
                        ? null
                        : _store.IndexBlockHash(chainId, lowestIndex - 1);
                    var highestBlockHash = _store.IndexBlockHash(chainId, highestIndex);

                    stateRefs = _store.ListAllStateReferences(
                        chainId,
                        lowestBlockHash,
                        highestBlockHash
                    );
```
Or we can remove the previous API.